### DISCUSSION TITLE
Can I replace the background image on the login page with a local file

### DISCUSSION THREAD

**Discussion by loveyu** [photoprism/photoprism#2622] [Q&A] [answered]
https://github.com/photoprism/photoprism/blob/61dfd2cf50b5eed436636c8528bd0bd7c74846dc/frontend/src/css/wallpapers.css#L10

https://cdn.photoprism.app/wallpaper/welcome.jpg

loading too slow....

![image](https://user-images.githubusercontent.com/1912373/185648094-5508d1fb-baf1-4b6b-9fb8-9721bdbf4a14.png)

**Comment by graciousgrey** [comment, marked as answer]
Yes, sponsors can change the background image

https://docs.photoprism.app/getting-started/config-options/#customization

https://photoprism.app/editions

**Comment by UkieHub** [comment]
![SmartSelect_20230823_175059_Chrome](https://github.com/photoprism/photoprism/assets/20022344/7c9d1c32-fa5f-4fee-b6ed-37ccdec45a26)
I have the same problem. I have to changing URI background, wallpaper in login is just black.
config output
| wallpaper-uri          | /static/img/wallpaper/w.jpg |

**Reply by lastzero** [reply]
You need a complete URL with protocol and hostname.

**Reply by UkieHub** [reply]
I Already tried that. But it did not fix the problem.
Running a CLI command I get this - 

root@NAS:~# docker exec -ti photoprism photoprism config  wallpaper-uri https://nas.ukiehub.com/apps/files_sharing/publicpreview/K9F7W8FnQr3pH8M?file=/&fileId=262279&x=1081&y=2317&a
[1] 15406
[2] 15407
[3] 15408
[4] 15409
-bash: a: command not found
[2]   Done                    fileId=262279
[3]-  Done                    x=1081
[4]+  Done                    y=2317

**Reply by lastzero** [reply]
```
docker exec -ti photoprism photoprism --wallpaper-uri="https://..." config
```

**Reply by UkieHub** [reply]
Still nothing
After running a CLI
admin@NAS:~/photoprism$ docker exec -ti photoprism photoprism --wallpaper-uri="https://nas.ukiehub.com/apps/files_sharing/publicpreview/K9F7W8FnQr3pH8M?file=/&fileId=262279&x=1081&y=2317&a" config
It did change  in config
wallpaper-uri          | https://nas.ukiehub.com/apps/files_sharing/publicpreview/K9F7W8FnQr3pH8M?file=/&fileId=262279&x=1081&y=2317&a |

But after stoping the container and starting again, clearing cache in chrome running a 
docker exec -ti photoprism photoprism config
It would go back to
| wallpaper-uri          | /static/img/wallpaper/welcome.jpg |
With no wallpaper-uri  in config.yml
Only after adding 
PHOTOPRISM_WALLPAPER_URI: "https://nas.ukiehub.com/apps/files_sharing/publicpreview/K9F7W8FnQr3pH8M?file=/&fileId=262279&x=1081&y=2317&a=true"
To docker-compose.yml it would change the config output.
But the wallpaper on the login would still be black.

**Reply by lastzero** [reply]
Could be due to wrong/missing encoding of special characters like / in the query parameters. Note that we are out of the office for a week and can only offer limited support.

**Reply by UkieHub** [reply]
Thanks for any help that you can provide.
This is my all output of config

GLOBAL CONFIG OPTIONS
|------------------------|--------------------------------------------------------------------------------------------------------------------|
|          Name          |                                                       Value                                                        |
|------------------------|--------------------------------------------------------------------------------------------------------------------|
| auth-mode              | password                                                                                                           |
| admin-user             | admin                                                                                                              |
| admin-password         | ***********                                                                                                        |
| public                 | false                                                                                                              |
| session-maxage         |                                                                                                            1209600 |
| session-timeout        |                                                                                                             604800 |
| login-uri              | /library/login                                                                                                     |
| register-uri           |                                                                                                                    |
| password-length        |                                                                                                                  8 |
| password-reset-uri     |                                                                                                                    |
| log-level              | debug                                                                                                              |
| debug                  | true                                                                                                               |
| trace                  | false                                                                                                              |
| config-path            | /photoprism/storage/config                                                                                         |
| certificates-path      | /photoprism/storage/config/certificates                                                                            |
| options-yaml           | /photoprism/storage/config/options.yml                                                                             |
| defaults-yaml          | /etc/photoprism/defaults.yml                                                                                       |
| settings-yaml          | /etc/photoprism/settings.yml (defaults)                                                                            |
| settings-yaml          | /photoprism/storage/config/settings.yml                                                                            |
| originals-path         | /photoprism/originals                                                                                              |
| originals-limit        |                                                                                                              50000 |
| resolution-limit       |                                                                                                                150 |
| users-path             | users                                                                                                              |
| users-originals-path   | /photoprism/originals/users                                                                                        |
| storage-path           | /photoprism/storage                                                                                                |
| users-storage-path     | /photoprism/storage/users                                                                                          |
| sidecar-path           | /photoprism/storage/sidecar                                                                                        |
| albums-path            | /photoprism/storage/albums                                                                                         |
| backup-path            | /photoprism/storage/backup                                                                                         |
| cache-path             | /photoprism/storage/cache                                                                                          |
| cmd-cache-path         | /photoprism/storage/cache/cmd                                                                                      |
| media-cache-path       | /photoprism/storage/cache/media                                                                                    |
| thumb-cache-path       | /photoprism/storage/cache/thumbnails                                                                               |
| import-path            | /photoprism/import                                                                                                 |
| import-dest            |                                                                                                                    |
| assets-path            | /opt/photoprism/assets                                                                                             |
| static-path            | /opt/photoprism/assets/static                                                                                      |
| build-path             | /opt/photoprism/assets/static/build                                                                                |
| img-path               | /opt/photoprism/assets/static/img                                                                                  |
| templates-path         | /opt/photoprism/assets/templates                                                                                   |
| temp-path              | /tmp/photoprism_9fca11fd                                                                                           |
| workers                |                                                                                                                  1 |
| wakeup-interval        | 15m0s                                                                                                              |
| auto-index             |                                                                                                                120 |
| auto-import            |                                                                                                                150 |
| read-only              | false                                                                                                              |
| experimental           | true                                                                                                               |
| disable-webdav         | false                                                                                                              |
| disable-settings       | false                                                                                                              |
| disable-places         | false                                                                                                              |
| disable-backups        | false                                                                                                              |
| disable-tensorflow     | false                                                                                                              |
| disable-faces          | false                                                                                                              |
| disable-classification | false                                                                                                              |
| disable-sips           | true                                                                                                               |
| disable-ffmpeg         | false                                                                                                              |
| disable-exiftool       | false                                                                                                              |
| disable-darktable      | false                                                                                                              |
| disable-rawtherapee    | false                                                                                                              |
| disable-imagemagick    | false                                                                                                              |
| disable-heifconvert    | false                                                                                                              |
| disable-rsvgconvert    | false                                                                                                              |
| disable-vectors        | false                                                                                                              |
| disable-jpegxl         | false                                                                                                              |
| disable-raw            | false                                                                                                              |
| raw-presets            | false                                                                                                              |
| exif-bruteforce        | false                                                                                                              |
| detect-nsfw            | false                                                                                                              |
| upload-nsfw            | true                                                                                                               |
| tensorflow-version     | 1.15.2                                                                                                             |
| tensorflow-model-path  | /opt/photoprism/assets/nasnet                                                                                      |
| default-locale         | en                                                                                                                 |
| default-theme          | default                                                                                                            |
| app-name               | UkieHub                                                                                                            |
| app-mode               | standalone                                                                                                         |
| app-icon               | logo                                                                                                               |
| app-color              | #000000                                                                                                            |
| legal-info             |                                                                                                                    |
| legal-url              |                                                                                                                    |
| wallpaper-uri          | https://nas.ukiehub.com/apps/files_sharing/publicpreview/K9F7W8FnQr3pH8M?file=/&fileId=262279&x=1081&y=2317&a=true |
| cdn-url                | /                                                                                                                  |
| cdn-video              | false                                                                                                              |
| site-url               | https://photos.ukiehub.com/                                                                                        |
| site-https             | true                                                                                                               |
| site-domain            | photos.ukiehub.com                                                                                                 |
| site-author            |                                                                                                                    |
| site-title             | UkieHub                                                                                                            |
| site-caption           | Ukiehub Photos 🇺🇦                                                                                                   |
| site-description       | PhotoCloud library                                                                                                 |
| site-preview           | https://i.photoprism.app/prism?cover=64&style=centered%20dark&caption=none&title=UkieHub                           |
| base-uri               | /                                                                                                                  |
| api-uri                | /api/v1                                                                                                            |
| static-uri             | /static                                                                                                            |
| content-uri            | /api/v1                                                                                                            |
| video-uri              | /api/v1                                                                                                            |
| https-proxy            |                                                                                                                    |
| https-proxy-insecure   | false                                                                                                              |
| trusted-proxy          | 172.16.0.0/12                                                                                                      |
| proxy-proto-header     | X-Forwarded-Proto                                                                                                  |
| proxy-proto-https      | https                                                                                                              |
| disable-tls            | true                                                                                                               |
| default-tls            | false                                                                                                              |
| tls-email              |                                                                                                                    |
| tls-cert               |                                                                                                                    |
| tls-key                |                                                                                                                    |
| http-mode              | debug                                                                                                              |
| http-compression       | gzip                                                                                                               |
| http-cache-maxage      |                                                                                                            2592000 |
| http-cache-public      | false                                                                                                              |
| http-host              | 0.0.0.0                                                                                                            |
| http-port              |                                                                                                               2342 |
| database-driver        | mysql                                                                                                              |
| database-name          | photoprism                                                                                                         |
| database-server        | mariadb:3306                                                                                                       |
| database-host          | mariadb                                                                                                            |
| database-port          |                                                                                                               3306 |
| database-user          | photoprism                                                                                                         |
| database-password      | ********                                                                                                           |
| database-conns         |                                                                                                                 20 |
| database-conns-idle    |                                                                                                                 10 |
| mariadb-bin            | /usr/bin/mariadb                                                                                                   |
| mariadb-dump-bin       | /usr/bin/mariadb-dump                                                                                              |
| sips-bin               |                                                                                                                    |
| sips-blacklist         | avif,avifs                                                                                                         |
| ffmpeg-bin             | /usr/bin/ffmpeg                                                                                                    |
| ffmpeg-encoder         | libx264                                                                                                            |
| ffmpeg-size            |                                                                                                               3840 |
| ffmpeg-bitrate         |                                                                                                                 50 |
| ffmpeg-map-video       | 0:v:0                                                                                                              |
| ffmpeg-map-audio       | 0:a:0?                                                                                                             |
| exiftool-bin           | /usr/bin/exiftool                                                                                                  |
| darktable-bin          | /usr/bin/darktable-cli                                                                                             |
| darktable-cache-path   |                                                                                                                    |
| darktable-config-path  |                                                                                                                    |
| darktable-blacklist    |                                                                                                                    |
| rawtherapee-bin        | /usr/bin/rawtherapee-cli                                                                                           |
| rawtherapee-blacklist  | dng                                                                                                                |
| imagemagick-bin        | /usr/bin/convert                                                                                                   |
| imagemagick-blacklist  | heif,heic,heics,avif,avifs,jxl                                                                                     |
| heifconvert-bin        | /usr/local/bin/heif-convert                                                                                        |
| rsvgconvert-bin        | /usr/bin/rsvg-convert                                                                                              |
| jpegxldecoder-bin      | /usr/bin/djxl                                                                                                      |
| download-token         | 2i53mib8                                                                                                           |
| preview-token          | 9fca11fd                                                                                                           |
| thumb-color            | sRGB                                                                                                               |
| thumb-filter           | lanczos                                                                                                            |
| thumb-size             |                                                                                                               7680 |
| thumb-size-uncached    |                                                                                                               7680 |
| thumb-uncached         | true                                                                                                               |
| jpeg-quality           |                                                                                                                 90 |
| jpeg-size              |                                                                                                              26240 |
| png-size               |                                                                                                               7680 |
| face-size              |                                                                                                                 50 |
| face-score             |                                                                                                           9.000000 |
| face-overlap           |                                                                                                                 42 |
| face-cluster-size      |                                                                                                                 80 |
| face-cluster-score     |                                                                                                                 15 |
| face-cluster-core      |                                                                                                                  4 |
| face-cluster-dist      |                                                                                                           0.640000 |
| face-match-dist        |                                                                                                           0.460000 |
| pid-filename           | /photoprism/storage/photoprism.pid                                                                                 |
| log-filename           | /photoprism/storage/photoprism.log                                                                                 |
| custom-assets-path     | /photoprism/plus/assets                                                                                            |
| custom-static-uri      | /c/static                                                                                                          |
|------------------------|--------------------------------------------------------------------------------------------------------------------|
WEB SECURITY
|----------------------|--------------------------------------------------------------------------------------------------------------|
|         Name         |                                                    Value                                                     |
|----------------------|--------------------------------------------------------------------------------------------------------------|
| http-csp             | frame-ancestors 'none'; default-src 'self' 'unsafe-inline' data: blob: wss: *.photoprism.app *.maptiler.com; |
| http-cto             | nosniff                                                                                                      |
| http-coop            | same-origin                                                                                                  |
| http-xss-protection  | 1; mode=block                                                                                                |
| http-referrer-policy | same-origin                                                                                                  |
| http-frame-options   | DENY                                                                                                         |
| http-hostname        |                                                                                                              |
| http-sts             |                                                                                                              |
| sts-seconds          |                                                                                                     63072000 |
| sts-subdomains       | false                                                                                                        |
| sts-preload          | false                                                                                                        |
| disable-sts          | true                                                                                                         |
| login-limit          |                                                                                                           10 |
| login-interval       | 1m0s                                                                                                         |
| ips-limit            |                                                                                                            3 |
| ips-interval         | 1h0m0s                                                                                                       |
|----------------------|--------------------------------------------------------------------------------------------------------------|

**Reply by UkieHub** [reply]
![SmartSelect_20230824_120912_Chrome](https://github.com/photoprism/photoprism/assets/20022344/c3eaed4e-dabf-482c-96fd-c987e1bacd52)
By the way if I chang 
PHOTOPRISM_WALLPAPER_URI: "https://cdn.photoprism.app/wallpaper/welcome.jpg" in docker-compse.yml to your old welcome.jpg it does change the login wallpaper image

**Reply by lastzero** [reply]
If you are loading the image from a different URL, you will additionally change the content security policy accordingly:
https://docs.photoprism.app/getting-started/config-options/#web-server

**Reply by UkieHub** [reply]
![SmartSelect_20230824_133754_Chrome](https://github.com/photoprism/photoprism/assets/20022344/6b381d08-9f2d-4835-a88d-7620946b14ec)
All got it fixed after creating a local link at https://photos.ukiehub.com and adding to config.yml
WallpaperUri: https://photos.ukiehub.com

**Reply by joemathis** [reply]
@UkieHub 
I'm kind of a container idiot, and can't figure out how to create a local link for the website.  Are you adding the file into a volume and then putting that as the URI in the docker-compose.yml file?

**Comment by UkieHub** [comment]
![SmartSelect_20230831_105141_Chrome](https://github.com/photoprism/photoprism/assets/20022344/ca91655c-a081-4237-b90e-bbe587b1a82b)
/I
![SmartSelect_20230831_105239_Chrome](https://github.com/photoprism/photoprism/assets/20022344/cf051afa-3193-42f7-9f58-f8d0559bf83a)
 uploaded the image to my library then opened in my browser us just an image. The URI I put it in
 photoprism/storage/config/options.yml

WallpaperUri: https://photos.ukiehub.com/api/v1/t/ad35aeb
Or what ever link you will have.

You can try to put in docker_compose.yml as
PHOTOPRISM_WALLPAPER_URI: "https://.........." As well you can try.

After all that just restart the container from ssh
docker_compose stop
docker_compose up -d
Run a command
docker exec -ti photoprism photoprism config
|app-name               | UkieHub                                                                                        |
| app-mode               | standalone                                                                                     |
| app-icon               | /static/icons/logo.svg                                                                         |
| app-color              | #000000                                                                                        |
| legal-info             |                                                                                                |
| legal-url              |                                                                                                |
| wallpaper-uri          | https://photos.ukiehub.com/api/v1/t/ad35aeb7d09aa57148dee8983db4c16728 |
| cdn-url                | /                                                                                              |
| cdn-video              | false                                                                                          |
| site-url               | https://photos.ukiehub.com/                                                                    |
See if it's there.
Clear the browser cash. When the first time you try to log in the image would not be there.
Log in, then logout and you should see the wallpaper updated.
I hope I helped in any way.

**Reply by joemathis** [reply]
It looks great and that did set the wallpaper on the login screen.
Thank you for that.

Does this actually show any preview in browsers? 
| site-preview | https://i.photoprism.app/prism?cover=64&style=centered%20dark&caption=none&title=UkieHub |

**Reply by UkieHub** [reply]
Yes you can changed in options.yml
SitePreview: https://photos.ukiehub.com/api/v1/t/ad35aeb7d09aa57148dee8983db4c1672882039c/1b1xz0bm/fit_1280

and it will change in your config output 
| site-caption           | Ukiehub Photos 🇺🇦                                                                               |
| site-description     | PhotoCloud library                                                                                |
| site-preview           | https://photos.ukiehub.com/api/v1/t/ad35aeb7d09aa57148dee8983db4c1672882039c/1b1xz0bm/fit_1280 |

But where or how do use see or access that site preview in your browser ?

**Reply by joemathis** [reply]
I'm using the URL in facebook.  posting pictures of sons football games to the parents group and they don't follow the PhotoPrism splash screen that comes up.  Too paranoid about clicking phishing links I guess.  So I wanted it to be something they know or can relate to.
